12-watchtower-API.md

```diff
@@ -22,227 +22,320 @@ The scope of this bolt does not include:
+
+## Quality of Service data
+
+`qos_data` is a list where each field specifies they type and associated data of the offered/requested `qos`. The format is defined as follows:
+
+* [`u16`: `qos_type`]
+* [`u16`: `data_len`]
+* [`data_len*byte`: `data`]
+
+So far, only `accountability` is defined.
+
+### `accountability`
+
+The accountability `qos` defines a pair `qos_data` blobs, associated to a pair of messages: The first one is `customer_evidence` and it is provided by the `client` in the `appointment` message. The second one it is `tower_evidence`, and is provided by the WatchTower in the `appointment_accepted` message.
+
+#### `customer_evidence`
+
+The format for the `customer_evidence` is defined as follows:
 
+1. type: ? (`customer_evidence`)
+2. data:  
+	* [`u64 `:`dispute_delta`]
+	* [`u64`: `transaction_size`]
+	* [`u64`: `transaction_fee`]
 
+If `accountability` is being requested, the sending node:
 	
+* MUST set `dispute_delta` to the CLTV value specified in the `commitment_transaction`.
+* MUST set `transaction_size` to the size of the serialized `justice_transaction`, in bytes.
+* MUST set `transaction_fee` to the fee set in the `justice_transaction`, in satoshis.
+* MUST set the `customer_signature_algorithm` to one of the signature algorithms supported by the tower.
+* MUST set `customer_signature` to the signature of the appointment using `op_customer_signature_algorithm`.
+* MUST set `customer_public_key` to the public key that matches the private key used to create `op_customer_signature`.
+
+If `accountability` is being offered, the receiving node:
+
+* MUST compute the `customer_signature` verification using `customer_public_key`.
+* SHOULD compute the `fee_rate` set in the `justice_tx` using `transaction_size` and `transaction_fee`.
+
+* MUST reject the appointment if:
+	* Any of the fields is missing.
+	* `transaction_size` is unreasonable.
+	* `customer_signature_algorithm` does not match any of the supported signing algorithms.
+	* `customer_signature` cannot be verified using `customer_public_key`.
+
+* SHOULD reject the appointment if:
+	* `dispute_delta` is too small.
+	* `fee_rate` is too low.
+
+If `accountability` is NOT being offered:
+
+* The receiving node MUST reject the appointment.
+
+Otherwise:
+
+* The receiving node SHOULD accept the appointment.
+
 #### Rationale
-The transaction `locator` can be deterministically computed by both the client and the server. Locators of wrong size are therefore invalid.
 
-`start_block` and `end_block` too close to the current chain tip may result in the tower missing the trigger and therefore should be avoided.
+The concept of too small for `dispute_delta` is subjective. The `dispute_delta` defines the time (in blocks) that the tower has to respond after a breach is seen. The smallest the value, the more the server risks to fail the appointment.
```

Comment:
Minor suggestions:

"the time (in blocks) that the tower has to respond" => "the time (in blocks) that the tower has _in order_ to respond"
"The smallest the value" =>  "The _smaller_ the value"